Assets/Scripts/AbstractClasses/IItemHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Crafting;
+using Inventory;
+
+namespace Interfaces
+{
+    /// <summary>
+    /// Everything that stores items. Especially used in <see cref="CraftingRecipe"/> for crafting.
+    /// </summary>
+    public interface IItemHandler
+    {
+        /// <summary>
+        /// Event that will be triggered after an item is changed/added/removed
+        /// </summary>
+        event Action ItemsUpdated;
+
+        /// <summary>
+        /// Checks if the given item is present (in given quantity)
+        /// </summary>
+        /// <param name="item">The item to be checked</param>
+        /// <param name="amount">The amount of this item to be present</param>
+        /// <returns>Whether the item and the given amount is present</returns>
+        bool ContainsItem(Item item, int amount = 1);
+
+        /// <summary>
+        /// Adds an item with a specified quantity
+        /// </summary>
+        /// <param name="item">The item to be added</param>
+        /// <param name="amount">The amount to be added</param>
+        void AddItem(Item item, int amount);
```

Comment:
In my opinion this method should also define a default value for the amount of items, for consistency reasons :)